atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchManagerImpl.java

```diff
@@ -27,10 +32,15 @@
 
 public final class LockWatchManagerImpl implements LockWatchManager {
     private final NamespacedLockWatchingRpcClient lockWatchingRpcClient;
+    private final NamespacedConjureTimelockService timelock;
     private final LockWatchEventCache cache;
 
-    public LockWatchManagerImpl(NamespacedLockWatchingRpcClient lockWatchingRpcClient, LockWatchEventCache cache) {
+    public LockWatchManagerImpl(
```

Comment:
I think this interface is weird to be putting on TransactionManager - we've tried hard to shy away from exposing yet more internal implementation details of AtlasDB to the world.

At most, clients need access to registerWatches - the other two methods should remain somewhere where they aren't available. Maybe getEventsForTransactions and getCommitUpdate could be package private? Or otherwise in a different class, where only registerWatches is visible here?

Reply:
The commit update can be hidden as it's only used by the TM internally, but we need some kind of interface for clients to register watches and get the watches for their transactions if they are to use this information to implement caching themselves.

I am open to suggestions on how to handle it better, but package private is not going to work and exposing it at the transaction level is even worse.

Reply:
Proposed solution here: https://github.com/palantir/atlasdb/pull/4799